src/pyqasm/visitor.py

```diff
@@ -2481,6 +2521,43 @@ def _evaluate_case(statements):
             default_stmts = statement.default.statements
             return _evaluate_case(default_stmts)
 
+    def _visit_delay_statement(
+        self, statement: qasm3_ast.DelayInstruction
+    ) -> list[qasm3_ast.Statement]:
+        _delay_time_var = statement.duration
+        global_scope = self._get_global_scope()
+        # validate the time variable assigned to delay operation
+        PulseValidator.validate_delay_time_var(_delay_time_var, statement, global_scope)
+        duration_val, _ = Qasm3ExprEvaluator.evaluate_expression(
```

Comment:
Should we not validate that the evaluated duration value is >= 0? I don't think there will be an application for negative time durations in the `delay` operation